src/main/java/org/ow2/proactive/catalogclient/service/CatalogResolveObjectControllerApi.java

```diff
@@ -115,6 +87,16 @@ public String getResolvedCatalogObject(String catalogUrl, String bucketName, Str
         return resource;
     }
 
+    private String getCatalogObjectAsString(String bucketName, String objectName, String sessionId) {
+        File file = catalogObjectControllerApi.getRawUsingGET(bucketName, objectName, sessionId);
+        try {
+            return Files.toString(file, Charsets.UTF_8);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)